primefaces/src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java

```diff
@@ -138,7 +141,7 @@ public void filter(FacesContext context, DataTable table) {
 
                 if (globalFilter != null && globalFilter.isActive() && !globalMatch.get() && !hasGlobalFilterFunction) {
                     FilterConstraint constraint = globalFilter.getConstraint();
-                    Object filterValue = globalFilter.getFilterValue();
+                    Object filterValue = LangUtils.normalize(globalFilter.getFilterValue(), table.isFilterNormalize());
                     globalMatch.set(constraint.isMatching(context, columnValue, filterValue, filterLocale));
```

Comment:
We should normalize the columnValue as well, right?

Reply:
Good point. We probably should for comparison.

Reply:
@melloware basically this was the same issue I believe.